### PR TITLE
feat(spanner): set client wide ReadOptions, ApplyOptions, and TransactionOptions

### PR DESCRIPTION
**Pull request by shuheiktgw**
Currently, if we want to set ReadOptions, ApplyOptions, or TransactionOptions, we need to pass those options as a parameter each time we call the related functions. That is tedious and hard to maintain. Instead, I'd like to specify those options as part of the ClientConfig and use them as default values as the QueryOptions already does. Thanks for your review 🙂

### CODE REVIEW

File: spanner/transaction.go [hunk 1]
```diff
@@ -1106,11 +1143,19 @@ type CommitResponse struct {
...
+// CommitOptions provides options for committing a transaction in a database.
 type CommitOptions struct {
 	ReturnCommitStats bool
 }
 
+// merge combines two CommitOptions that the input parameter will have higher
+// order of precedence.
+func (co CommitOptions) merge(opts CommitOptions) CommitOptions {
+	return CommitOptions{
+		ReturnCommitStats: co.ReturnCommitStats || opts.ReturnCommitStats,
```
Comment (shuheiktgw) [inline spanner/transaction.go:1155]: I could not figure out a good way to merge the ReturnCommitStats...

**Comment by shuheiktgw** [comment]
@rahul2393 Would you review this PR when you have time? 🙂

**Comment by rahul2393** [comment]
@shuheiktgw Can you also please add relevant tests for this?

**Comment by shuheiktgw** [comment]
Thanks for your comment, @rahul2393! I've added unit tests, so would you review the PR? 🙏

**Comment by rahul2393** [comment]
Please resolve conflicts @shuheiktgw

**Comment by shuheiktgw** [comment]
Sure, I've resolved the conflict so would you take a look? 🙂 @rahul2393

**Review by rahul2393 — APPROVED** [review]

**Pull request merged**